executor/api/kafka/server.go

```diff
@@ -101,7 +102,17 @@ func NewKafkaServer(fullGraph bool, workers int, deploymentName, namespace, prot
 				producerConfigMap["ssl.certificate.pem"] = sslKakfaServer.ClientCert
 			}
 			producerConfigMap["ssl.key.password"] = sslKakfaServer.ClientKeyPass // Key password, if any
-
+			if util.GetKafkaSecurityProtocol() == "SASL_SSL" {                   //if we also have SASL enabled, then we need to provide the necessary settings in addition to SSL
+				producerConfigMap["sasl.mechanisms"] = saslKafkaServer.Mechanism
+				if saslKafkaServer.UserNameFile != "" && saslKafkaServer.UserPasswordFile != "" {
+					producerConfigMap["sasl.username"] = saslKafkaServer.UserNameFile
```

Comment:
:question: Does using files instead of strings work for username and password?  I can't see how the underlying Kafka library would differentiate the intents, but happy to be pointed to docs!